Laravel 6 support

MatanYadaev opened the pull request:
Closes https://github.com/tymondesigns/jwt-auth/issues/1860

MatanYadaev committed:
update composer.json

ajpsp5 commented:
@MatanYadaev Any ideas on how to get this to pass all test?

shirshak55 commented:
only bumping version will support laravel 6?

specialtactics commented:
Why are versions <7 even in travis, they haven't been supported for a very long time. Maybe I will make a PR to remove them.

Not sure why 7.3 is failing though :thinking: you should investigate

specialtactics commented:
I believe the develop branch is currently broken - if you check out my PR here, it made also no changes other than to remove old versions, and the exact same test fails

https://github.com/tymondesigns/jwt-auth/pull/1864

specialtactics commented:
Ok I found out what was happening to that test, fixed it in my PR (above), once merged, you should sync with develop and this PR will pass.

MatanYadaev committed:
fix test using DateInterval wrong

MatanYadaev committed:
remove code inserted by mistake to composer.json

MatanYadaev committed:
fix travis to use trusty images for old php versions; remove php 5.5 from tests, only 5.5.9 supported

MatanYadaev committed:
fix previous commit - mistake in .travis.xml

MatanYadaev committed:
laravel 6.0 test

MatanYadaev commented:
- Fixed `it_should_handle_datetinterval_claims` test
- Fixed TravisCI to use Trusty images instead of Xenial for PHP 5.5 & 5.6
- Removed PHP 5.5, we already testing PHP 5.5.9
- Added Laravel 6.0 to the tests (PHP 7.2 & PHP 7.3 with coverage)

MatanYadaev committed:
add laravel 6.0 to test with php 7.2

MatanYadaev commented:
Everything looks OK 😄 Only need to wait for `cartalyst/sentinel` to release their version for Laravel 6.0: https://github.com/cartalyst/sentinel/issues/486

BTW for some reason, 2 environments on Travis passed the 16 minutes build and failed, any idea why it happened or how to retry them?

tymondesigns commented:
@MatanYadaev Thank you for your efforts on this! I made the changes directly to develop and worked around the sentinel issues

aligajani commented:
@MatanYadaev  @tymondesigns  Thanks guys. Should it work on Laravel v6 now by doing `composer require ` ? Which documentation should I follow if any?

shirshak55 commented:
@aligajani just set it to develop branch by changing composer.json and it should work.. And I think most of us should fork it and use it just like many corporation does in case there are other problem.

aligajani commented:
Thanks. So you’ve noticed no problem at all with Laravel 6 and this
package? I believe I can upgrade it to 1.0 when it does arrive but I’ll
stick with your advice for now.

On Thu, Sep 12, 2019 at 10:47 AM Shirshak <notifications@github.com> wrote:

> @aligajani <https://github.com/aligajani> just set it to develop branch
> by changing composer.json and it should work.. And I think most of us
> should fork it and use it just like many corporation does in case there are
> other problem.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tymondesigns/jwt-auth/pull/1861?email_source=notifications&email_token=AANGRWV4L3P44KIHTOTA4KTQJIF3RA5CNFSM4IPBNB5KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD6RJ6MQ#issuecomment-530751282>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AANGRWVOD75X5K6EOHDIMUDQJIF3RANCNFSM4IPBNB5A>
> .
>
-- 


Ali Gajani
Founder at Mr. Geek
www.mrgeek.me
www.aligajani.com

deefour commented:
@aligajani  I've had zero issues with Laravel 6 and my `composer.json` pointing at:

```json
"tymon/jwt-auth": "dev-develop"
```

aligajani commented:
Thank you guys. Another quick one, but unrelated to this thread: how do I use the laravel 6 auth scaffolding as is but with JWT auth? I mean I am assuming in an ideal setup it will not require changes at all and can be manipulated via API?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>
